src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,59 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, S> {
+    T first;
+    S second;
 
+    private Pair(T first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    public static <T, S> Pair of(T first, S second) {
+        return new Pair(first, second);
+
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        Pair<T, S> other = (Pair) obj;
```

Comment:
what happens if obj that you're passing here is String or Integer or any other class except Pair?

Reply:
added the check for obj class.